### ISSUE TITLE
Stacking set of dm4-files: Metadata-Title is no String

### ISSUE THREAD

**Issue by mgrove-wwu** [hyperspy/hyperspy#1937]
Hello again,
i hope that you guys are able to help again.
I have a problem getting around the problem shown below.

`im = hs.load('*.dm4', stack=True)`

`TypeError: must be str, not list`

Am i able to change the metadata in the files before loading them into hyperspy as a stack or perhaps specifying some sort of signal_type-list during loading? I guess its not possible and i have to change the files metadata before loading them as a stack, but i thought asking would not hurt.
Here a single image is loaded and the base info is given to:
`im` results in
`<Signal2D, title: [], dimensions: (|1920, 1792)>`
As you can see, the title of the files seems not to be a string but a list actually as the error states.
Why that is, i dont know, because i am not the experimentalist working on our TEM but just getting the data.
though these are just images, i will have to operate on images with a subset of EELS spectra.
Perhaps i missed some part in the documentary again, though i RTFM.

Error message (using anaconda): 
**TypeError Traceback (most recent call last)
<ipython-input-38-745a65e4cb4b> in <module>()
      1 im = hs.load("*.dm4", stack=True)
      2 im

/Apps/anaconda3/envs/HS/lib/python3.6/site-packages/hyperspy/io.py in load(filenames, signal_type, stack, stack_axis, new_axis_name, lazy, **kwds)
    231                 signal = stack_method(
    232                     signal, axis=stack_axis, new_axis_name=new_axis_name,
    233                     lazy=lazy)
    234                 signal.metadata.General.title = os.path.split(
    235                     os.path.split(os.path.abspath(filenames[0]))[0])[1]

/Apps/anaconda3/envs/HS/lib/python3.6/site-packages/hyperspy/misc/utils.py in stack(signal_list, axis, new_axis_name, lazy, **kwargs)
    864             # Get the title from 1st object
    865             signal.metadata.General.title = (
    866                 "Stack of " + first.metadata.General.title)
    867             signal.original_metadata = DictionaryTreeBrowser({})
    868         else:

TypeError: must be str, not list**

Thanks for taking the time to read this paragraph!

Sincerely,
Maximilian

PS: I heard that there is a possibility to save the files correctly with "tag-specification" in digital micrograph but do you guys know another workaround?

**Comment by magnunor** [comment]
I'm not certain how to fix the "title as list" issue, but you could try to load the data using:

```python
import glob
import hyperspy.api as hs
filename_list = glob.glob("*.dm4")
filename_list.sort()
signal_list = []
for filename in filename_list:
    s = hs.load(filename)
    signal_list.append(s)
s_stack = hs.stack(signal_list)
```

**Comment by ericpre** [comment]
We have seen something similar recently with other kind of metadata - see #1700. We couldn't find any file to reproduce this behaviour, so we could't test it properly. @pyth0nsn1ppe7, could you please share a share, so we can add better support for this kind of files? What is the DM version used to generate these files?

**Comment by mgrove-wwu** [comment]
Hi again @ericpre ,
i will ask tomorrow, what version it was taken with. And perhaps i can get further data, but what i know for now: It was taken as a stack in digital micrograph and afterwards saved in seperate files as for each second an average image was used. Perhaps this has something to do with the issue.

Following your link, i will look into it in more detail tomorrow and reply with additional data to the generation process, hopefully, too.

Thanks for help to both of you.

**Comment by mgrove-wwu** [comment]
I did not have the exact version, but from what i know it was from gatan micrograph suite v 2. something. Though it might take longer than i thought, maybe till next week, unfortunatly.

**Comment by mgrove-wwu** [comment]
Hello again and sorry for the late answer.

The files were taken by GMS v 2.34.888 or v 2.31.xxx . The camera used to for the images was a K2 - perhaps this is something in common with the other reports.
Sadly, this was the only information my colleagues could give.
Hope it helps.
Thanks for your support!

**Comment by ericpre** [comment]
@pyth0nsn1ppe7, would it be possible to share a file?

**Comment by mgrove-wwu** [comment]
Sure, this is the link to one of the files. Hope the linkage is by the rules.
https://ufile.io/wl3um

**Comment by francisco-dlp** [comment]
Fixed in #2026
